File: 2023/1/day1.py
import re


def main():
    # inputfile = "example1.txt"
    inputfile = "example2.txt"
    inputfile = "input.txt"
    with open(inputfile) as f:
        input = f.read()

    print(part2(input))


def part1(input):
    sum = 0
    for line in input.splitlines():
        digits = "".join(re.findall(r'\d+', line))
        firstdigit = digits[0]
        lastdigit = digits[-1]
        concat = firstdigit + lastdigit
        sum += int(concat)
        print(line, digits, firstdigit, lastdigit, concat, sum)
    return sum


def part2(input):
    sum = 0
    for line in input.splitlines():
        r = r'(\d|one|two|three|four|five|six|seven|eight|nine)'
        leftnumber = re.search(r, line).group()
        r = r'(\d|eno|owt|eerht|ruof|evif|xis|neves|thgie|enin)'
        rightnumber = re.search(r, line[::-1]).group()[::-1]
        leftnumber = replace_numbers(leftnumber)
        rightnumber = replace_numbers(rightnumber)
        concat = leftnumber + rightnumber
        sum += int(concat)
        print(line, leftnumber, rightnumber, concat, sum)
    return sum


def replace_numbers(string):
    string = string.replace("one", "1")
    string = string.replace("two", "2")
    string = string.replace("three", "3")
    string = string.replace("four", "4")
    string = string.replace("five", "5")
    string = string.replace("six", "6")
    string = string.replace("seven", "7")
    string = string.replace("eight", "8")
    string = string.replace("nine", "9")
    return string


main()

File: 2023/1/day1.js
function go() {
    var input = document.getElementById('input').value
    var lines = input.split('\n')
    var sum = 0
    for(var i = 0; i < lines.length; i++) {
        var digits = lines[i].replace(/\D/g, '')
        var digit1 = digits[0]
        var digit2 = digits[digits.length-1]
        var concat = parseInt(digit1 + digit2)
        sum += concat
        console.log(digit1, digit2, concat, sum)
    }
    alert(sum)
}

function go2() {
    var input = document.getElementById('input').value
    var lines = input.split('\n')
    var sum = 0
    for(var i = 0; i < lines.length; i++) {
        var digits = lines[i].replace('one', '1').replace('two', '2').replace('three', '3').replace('four', '4').replace('five', '5').replace('six', '6').replace('seven', '7').replace('eight', '8').replace('nine', '9').replace(/\D/g, '')
        var digit1 = digits[0]
        var digit2 = digits[digits.length-1]
        var concat = parseInt(digit1 + digit2)
        sum += concat
        console.log(digit1, digit2, concat, sum)
    }
    alert(sum)
}
